### PR TITLE
Only open the user project once per installation

### LINKED ISSUE #3474: Build fails if project target name contains space
Hi, as the subject, I have a problem when I compile the project after installing a dependency with cocoapods. The error is: "ld: library not found for -lPods-Project validation."
The problem seems to be derived from space in the target name. I have two targets and I need to add dependency for both.

I have last version of cocoapods.

Thanks

### LINKED ISSUE #4374: Pod install spends a lot of time on Xcodeproj::Project.open
🌈

Hi,

We have found that pod install spends a lot of time on Xcodeproj::Project.open. Since it is called multiple times with the same path, with a simple cache, we have managed to speed it up by 100%.

A sample of the speedup and how we achieved it is in this public fork of Braintree's sdk: https://github.com/danielribeiro/braintree_ios/commit/627f24202e2065266a76a05b5dc0f6b7b5588dfc#diff-4a25b996826623c4a3a4910f47f10c30

The profile output showing the time spend can be seen [here](http://danielribeiro.github.io/braintree_ios/ProfileForCallStackPrinter.html).

On one of our largest iOS repos, this 2x speedup amounts to a few minutes saved. I'd love to see this performance improvement be made more broadly available. Any thoughts?

### PR DESCRIPTION
**Pull request by segiddins**
Closes #4374 .

- [x] CHANGELOG
- [x] Documentation

\c @mrackwitz

### CODE REVIEW

File: lib/cocoapods/installer/analyzer/target_inspection_result.rb [hunk 1]
```diff
@@ -35,6 +35,8 @@ class TargetInspectionResult
         #         due to the presence of Swift source in the user's targets
         #
         attr_accessor :recommends_frameworks
+
+        attr_accessor :project
```
Comment (mrackwitz) [inline lib/cocoapods/installer/analyzer/target_inspection_result.rb]: Please document this method at least with the type of the accessor.

File: lib/cocoapods/installer/post_install_hooks_context.rb [hunk 2]
```diff
@@ -47,10 +50,14 @@ def self.generate(sandbox, aggregate_targets)
       # Pure data class which describes and umbrella target.
       #
       class UmbrellaTargetDescription
+        attr_accessor :user_project
```
Comment (mrackwitz) [inline lib/cocoapods/installer/post_install_hooks_context.rb]: Please add doc comment.

File: lib/cocoapods/target/aggregate_target.rb [hunk 3]
```diff
@@ -54,13 +54,14 @@ def podfile
     #
     attr_accessor :client_root
 
+    attr_accessor :user_project
```
Comment (mrackwitz) [inline lib/cocoapods/target/aggregate_target.rb]: Please add doc comment.

**Comment by mrackwitz** [comment]
Great work! (Ups, just saw the todo regardings docs… now we're on the safe side at least.)

**Comment by segiddins** [comment]
Will merge in the morning

File: lib/cocoapods/target/aggregate_target.rb [hunk 4]
```diff
@@ -77,11 +81,10 @@ def podfile
     #
     # @return [Array<PBXNativeTarget>]
     #
-    def user_targets(project = nil)
```
Comment (mrackwitz) [inline lib/cocoapods/target/aggregate_target.rb]: You didn't removed the parameter from the method documentation here.

**Pull request merged**